src/workers/newView.ts

```diff
@@ -24,12 +24,78 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
   return false;
 };
 
+const DEFAULT_TIMEZONE = 'UTC'; // in case user doesn't have a timezone set
+const INC_STREAK_QUERY = `
+WITH u AS (
+  SELECT "id",
+      /* Increment the current streak if
+        a) lastViewAt is NULL - this is a new user, we should start a new streak
+        b) the currentStreak is 0 - we should start a new streak
+        c) we didn't do it today already
+      */
+      ("lastViewAt" is NULL OR
+       "currentStreak" = 0 OR
+       DATE("lastViewAt" AT TIME ZONE COALESCE("timezone", $3)) <> DATE($2 AT TIME ZONE COALESCE("timezone", $3))
+      ) AS shouldIncrementStreak,
+
+      -- Increment maxStreak if it's the same as currentStreak
+      "currentStreak" = "maxStreak" AS shouldUpdateMaxStreak
```

Comment:
Just in case 😉 

```suggestion
      "currentStreak" >= "maxStreak" AS shouldUpdateMaxStreak
```

Reply:
Actually thinking about it, it's not even needed here. We can move this logic to the update section below